app/src/components/SectionPicker.tsx

```diff
@@ -7,46 +7,63 @@ import { getSearchByStringExample } from "@constants/frontend";
 
 interface Props {
     soc: SOC_Generic;
-    searchText: string;
-    setSearchText: (searchText: string) => void;
 }
 
 export default function SectionPicker(props: Props) {
     const [searchByString, setSearchByString] = useState<string>(
         getSearchByString(SearchBys[0]),
     );
+    const [searchText, setSearchText] = useState("");
     const searchBy = getSearchBy(searchByString);
 
     /* https://github.com/apollographql/apollo-client/issues/9583 */
     const [abortRef, setAbortRef] = useState(new AbortController());
-    const { isFetching, data: courses } = useQuery<Course[]>({
+    const {
+        isFetching,
+        data: courses,
+        refetch: fetchCourses,
+    } = useQuery<Course[]>({
         initialData: [],
-        queryKey: [props.searchText], // Re-fetch when searchText is updated
+        // queryKey: [props.searchText], // Re-fetch when searchText is updated
```

Comment:
Committing commented is bad practice.

We have to decide to make a change if it's better.

If you want to revert, we have version-control!

Reply:
I'll keep that in mind, thanks 😃